src/main.py

```diff
@@ -495,75 +475,68 @@ def silence_listener(throwaway_frames):
 	inp.close()
 
 
-def loop():
-	global player
+trigger_thread = None
+def trigger_callback(trigger_name):
+	global trigger_thread
 
-	while True:
-		record_audio = False
+	triggers.disable()
+
+	trigger_thread = threading.Thread(target=trigger_process, args=(trigger_name,))
+	trigger_thread.setDaemon(True)
+	trigger_thread.start()
 
-		# Enable reading microphone raw data
-		inp = alsaaudio.PCM(alsaaudio.PCM_CAPTURE, alsaaudio.PCM_NORMAL, config['sound']['input_device'])
-		inp.setchannels(1)
-		inp.setrate(16000)
-		inp.setformat(alsaaudio.PCM_FORMAT_S16_LE)
-		inp.setperiodsize(1024)
 
-		while not record_audio:
-			time.sleep(.1)
+def trigger_process(trigger_name):
+	global player
+
+	trigger = triggers.triggers[trigger_name]
```

Comment:
I know what your saying but i'd maybe do it a little different for reliability. 
in the trigger callback i'd check to see if it exists, extract it with this line, and pass the actual trigger into `trigger_process` when you start the thread. 
By this time you know everything will be fine. 
If the trigger isn't found (who knows why) you can return something meaningful etc during the callback. 

Just an idea, you don't have to.